### PR TITLE
Add CaseReference object, per spec, to our case schema/model

### PR DESCRIPTION
**Pull request by axmb**
I'll chat with Kelly next week, but for the time being, this adds the `CaseReference` object defined in the official line list spec to our data service schema/model. This is necessary to perform upsert for the purpose of deduplication, which I'll send subsequently.

I suspect we'll want to somewhat merge this with the `Sources` field; something for me to pick Kelly/Stephen's brain on next week.

### CODE REVIEW

**Comment by axmb** [comment]
[Implement upsert in our core data API](https://trello.com/c/SUEtqFml/223-implement-upsert-in-our-core-data-api)

File: data-serving/data-service/schemas/cases.schema.json [hunk 1]
```diff
@@ -9,6 +9,21 @@
...
         "bsonType": "int"
       },
+      "caseReference": {
+        "bsonType": "object",
+        "additionalProperties": false,
+        "properties": {
+          "_id": {
+            "bsonType": "objectId"
+          },
+          "dataSourceId": {
```
Comment (allysonjp715) [inline data-serving/data-service/schemas/cases.schema.json:19]: Will this be a foreign key to the _id from the sources collection? If so, do we need to keep the "sources" field in this schema? (Not suggesting deletion in this PR, should be it's own if necessary)

File: data-serving/data-service/schemas/cases.schema.json [hunk 2]
```diff
@@ -9,6 +9,21 @@
...
+            "bsonType": "string"
+          },
+          "dataEntryId": {
```
Comment (allysonjp715) [inline data-serving/data-service/schemas/cases.schema.json:22]: nit: I'm not sure if dataSourceId and dataEntryId are the most clear names. I looked at this PR before I read your deduping doc and was confused by what they represented. Maybe "sourceId" and "sourceEntryId"? Especially for dataEntryId I didn't understand that it was the id that came from the source.

File: data-serving/data-service/schemas/cases.schema.json (hunk 1 above)
Comment (axmb) [reply data-serving/data-service/schemas/cases.schema.json:19]: Yep, certainly a good question. This is related to my comment in the PR description re: `Sources`. Let's catch up with Kelly on this next week -- I'm not sure what her plans were for `Sources` now that the spec is finalized.

File: data-serving/data-service/schemas/cases.schema.json (hunk 2 above)
Comment (axmb) [reply data-serving/data-service/schemas/cases.schema.json:22]: Makes sense to me -- done.

**Pull request merged**